### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: anonymous-functions

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1711]
This issue describes how to implement the `<SLUG>` concept exercise for the <LANG> track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Learning objectives

- Be capable of defining a `lambda` function
- Know how to store a `lambda` in a variable and call it using `funcall` or `apply`
- Understand what it means for a `lambda` to be a "closure", capturing its environment

## Out of scope

- Variable scoping
- Local functions with `flet` or `labels`

## Concepts

- `anonymous-functions`
- `closures` (Need to add this concept to `concepts.csv`)

## Prerequisites

- `function-definition`
- `higher-order-functions`
- `variables`
- `expressions`

## Resources to refer to

- [Old CL Cookbook](http://cl-cookbook.sourceforge.net/functions.html)

### Hints

- [Closures](https://riptutorial.com/common-lisp/example/9815/closures)
- [Anonymous Functions](https://riptutorial.com/common-lisp/example/4119/defining-anonymous-functions)

### After

- [Fancy Lambdas](https://lepisma.xyz/2017/12/20/fancy-lambdas.html)

## Representer

Nothing yet

## Analyzer

- Enforce the use of `lambda` and disallow `defun`, `flet`, `labels`, etc

## Implementing

In addition to using `lambda`s as arguments to higher-order functions, it might be good to have the student implement basic currying here (returning anonymous functions from partially applied functions).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
